packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1461,14 +1484,72 @@ export class GarbageCollector implements IGarbageCollector {
 				);
 			} else {
 				nodeStateTracker.updateTracking(currentReferenceTimestampMs);
-				if (this.tombstoneMode && nodeStateTracker.state === UnreferencedState.SweepReady) {
-					const nodeType = this.runtime.getNodeType(nodeId);
-					if (nodeType === GCNodeType.DataStore || nodeType === GCNodeType.Blob) {
-						this.tombstones.push(nodeId);
-					}
+				if (nodeStateTracker.state === UnreferencedState.SweepReady) {
+					sweepReadyNodes.push(nodeId);
 				}
 			}
 		}
+
+		return sweepReadyNodes;
+	}
+
+	/**
+	 * Deletes nodes from both the runtime and garbage collection
+	 * @param sweepReadyNodes - nodes that are ready to be deleted
+	 */
+	private runSweepPhase(sweepReadyNodes: string[], gcData: IGarbageCollectionData) {
+		// TODO: GC:Validation - validate that removed routes are not double deleted
+		// TODO: GC:Validation - validate that the child routes of removed routes are deleted as well
+		const sweptRoutes = this.runtime.deleteUnusedNodes(sweepReadyNodes);
+		const updatedGCData = this.deleteSweptRoutes(sweptRoutes, gcData);
+		this.deleteSweptNodeTrackers(sweptRoutes);
+		this.trackDeletedNodes(sweptRoutes);
+		return updatedGCData;
+	}
+
+	/**
+	 * Removes nodes that are swept from GC state and updates the deleted nodes
+	 */
+	private deleteSweptNodeTrackers(gcNodes: string[]) {
+		for (const nodeId of gcNodes) {
+			const nodeStateTracker = this.unreferencedNodesState.get(nodeId);
+			// TODO: GC:Validation - assert that the nodeStateTracker is defined
+			if (nodeStateTracker !== undefined) {
+				// Stop tracking so as to clear out any running timers.
+				nodeStateTracker.stopTracking();
+				// Delete the node as we don't need to track it any more.
+				this.unreferencedNodesState.delete(nodeId);
+			}
+		}
+	}
+
+	/**
+	 * @returns IGarbageCollectionData after deleting the sweptRoutes from the gcData
+	 */
+	private deleteSweptRoutes(
+		sweptRoutes: string[],
+		gcData: IGarbageCollectionData,
+	): IGarbageCollectionData {
+		const sweptRoutesSet = new Set<string>(sweptRoutes);
+		const gcNodes: { [id: string]: string[] } = {};
+		for (const [id, outboundRoutes] of Object.entries(gcData.gcNodes)) {
+			if (!sweptRoutesSet.has(id)) {
+				gcNodes[id] = Array.from(outboundRoutes);
+			}
+		}
+
```

Comment:
Without whitespaces, its so hard to make sense of the formatting :-(.

Reply:
Not sure what you mean here, do you want me to add new lines somewhere?

Reply:
nope, just complaining :)